test/run_test.py

```diff
@@ -889,12 +889,20 @@ def main():
             print_to_stderr(err_message)
     finally:
         if options.coverage:
+            from coveage import Coverage
+            cwd = os.getcwd()
             test_dir = os.path.dirname(os.path.abspath(__file__))
-            if not PYTORCH_COLLECT_COVERAGE:
-                shell(['coverage', 'combine'], cwd=test_dir)
-                shell(['coverage', 'html'], cwd=test_dir)
-            else:
-                shell(['coverage', 'combine', '--append'], cwd=test_dir)
+            try:
+                os.chdir(test_dir)
```

Comment:
I really wish that `os.chdir` was a `contextmanager` so that we don't have to do this `try/finally` statement here.

Just a nit.

Reply:
if we use this pattern in multiple places then we could replace it with a `contextmanager` as defined here: https://stackoverflow.com/a/37996581/5044950

Reply:
yup. decorator pattern saves the day :-P

Reply:
Completely agree on the context manager, though about doing in in this PR, but than decided to keep those changes separate

Reply:
Here is the BE PR: https://github.com/pytorch/pytorch/pull/49657